src/main/java/frc/trigon/robot/poseestimation/apriltagcamera/AprilTagCamera.java

```diff
@@ -0,0 +1,173 @@
+package frc.trigon.robot.poseestimation.apriltagcamera;
+
+import edu.wpi.first.math.Matrix;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.geometry.*;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.numbers.N3;
+import frc.trigon.robot.Robot;
+import frc.trigon.robot.constants.FieldConstants;
+import frc.trigon.robot.poseestimation.poseestimator.PoseEstimator6328;
+import org.littletonrobotics.junction.Logger;
+import org.photonvision.PhotonUtils;
+
+/**
+ * A pose source is a class that provides the robot's pose, from a camera.
+ */
+public class AprilTagCamera {
+    protected final String name;
+    private final RobotPoseSourceInputsAutoLogged inputs = new RobotPoseSourceInputsAutoLogged();
+    private final Transform3d robotCenterToCamera;
+    private final AprilTagCameraIO aprilTagCameraIO;
+    private double lastUpdatedTimestamp;
+    private final double solvePNPTranslationsStdExponent, assumedRobotPoseTranslationsStdExponent, solvePNPThetaStdExponent;
+    private Pose2d robotPose = null;
+
+    public AprilTagCamera(AprilTagCameraConstants.RobotPoseSourceType robotPoseSourceType, String name, Transform3d robotCenterToCamera, double solvePNPTranslationsStdExponent, double assumedRobotPoseTranslationsStdExponent, double solvePNPThetaStdExponent) {
```

Comment:
Add javadoc and explain params more in depth
SolvePNP expos should be next to each other